keanu-project/src/test/java/io/improbable/keanu/vertices/AssertVertexTest.java

```diff
@@ -0,0 +1,104 @@
+
+    @Rule
+    public DeterministicRule rule = new DeterministicRule();
+
+    @Test
+    public void assertThrowsOnFalseConstBool() {
+        thrown.expect(AssertionError.class);
+        ConstantBoolVertex constBool = new ConstantBoolVertex(BooleanTensor.create(false));
+        constBool.assertTrue().eval();
+    }
+
+    @Test
+    public void assertPassesOnTrueConstBool() {
+        ConstantBoolVertex constBool = new ConstantBoolVertex(BooleanTensor.create(true));
+        constBool.assertTrue().eval();
+    }
+
+    @Test
+    public void lazyEvalThrowsOnFalseConstBool() {
+        thrown.expect(AssertionError.class);
+        ConstantBoolVertex constBool = new ConstantBoolVertex(BooleanTensor.create(false));
+        constBool.assertTrue().lazyEval();
+    }
+
+    @Test
+    public void assertPassesOnRVWithTruePredicate() {
+        UniformVertex uniform = new UniformVertex(0, 5);
+        BoolVertex predicate = uniform.lessThan(new ConstantDoubleVertex(new double[]{10}));
+        predicate.assertTrue().eval();
+    }
+
+    @Test
+    public void assertPassesOnRVWithFalsePredicate() {
+        thrown.expect(AssertionError.class);
+        UniformVertex uniform = new UniformVertex(0, 5);
+        BoolVertex predicate = uniform.greaterThan(new ConstantDoubleVertex(new double[]{10}));
+        predicate.assertTrue().eval();
+    }
+
+    @Test
+    public void samplingWithAssertionWorks() {
+        thrown.expect(AssertionError.class);
+        GaussianVertex gaussian = new GaussianVertex(5, 1);
+        gaussian.greaterThan(new ConstantDoubleVertex(1000)).assertTrue();
+
+        BayesianNetwork bayesianNetwork = new BayesianNetwork(gaussian.getConnectedGraph());
+        MetropolisHastings.withDefaultConfig().generatePosteriorSamples(bayesianNetwork, bayesianNetwork.getLatentVertices()).generate(10);
+    }
+
+    @Test
+    public void valuePropagationPassesOnTrueAssertion() {
+        ConstantDoubleVertex constDouble = new ConstantDoubleVertex(3);
+        constDouble.lessThan(new ConstantDoubleVertex(10)).assertTrue();
+        constDouble.setAndCascade(8);
+    }
+
+    @Test
+    public void valuePropagationThrowsOnFalseAssertion() {
+        thrown.expect(AssertionError.class);
+        ConstantDoubleVertex constDouble = new ConstantDoubleVertex(3);
+        constDouble.lessThan(new ConstantDoubleVertex(10)).assertTrue();
+        constDouble.setAndCascade(15);
+    }
+
+    @Test
+    public void optimizerWithAssertionWorks() {
+        thrown.expect(AssertionError.class);
+        UniformVertex temperature = new UniformVertex(20, 30);
+        GaussianVertex observedTemp = new GaussianVertex(temperature, 1);
+        observedTemp.observe(29);
+        temperature.greaterThan(new ConstantDoubleVertex(34)).assertTrue();
+        KeanuOptimizer.of(temperature.getConnectedGraph()).maxAPosteriori();
+    }
+
+    @Test
+    public void assertGivesCorrectErrorMessage() {
```

Comment:
could you please add tests for the `expectMessage` when:
* there is no message in the `AssertVertex`
* the `AssertVertex` has no label